### PR TITLE
Wait for the network to be online before starting tinc. (1.0)

### PR DESCRIPTION
**Pull request by fangfufu**
Basically the same as pull request https://github.com/gsliepen/tinc/pull/326, but for tinc 1.0.

### CODE REVIEW

**Comment by fangfufu** [comment]
@hg , I get these when I put the ``DynamicUser`` in:
```
Aug 17 12:12:24 gabriel systemd[1]: Started Tinc net tinc.
Aug 17 12:12:24 gabriel tincd[446858]: Couldn't write pid file /run/tinc.tinc.pid: Read-only file system
Aug 17 12:12:24 gabriel systemd[1]: tinc@tinc.service: Main process exited, code=exited, status=1/FAILURE
Aug 17 12:12:24 gabriel systemd[1]: tinc@tinc.service: Failed with result 'exit-code'.
```

**Comment by fangfufu** [comment]
@hg, I tried your kludge (https://github.com/gsliepen/tinc/pull/326#issuecomment-900203979) ``ExecStartPre=+/usr/bin/chown -R tinc:tinc @sysconfdir@/tinc/%i``, I get the followings:
```
Aug 17 12:34:05 gabriel systemd[12845]: tinc@tinc.service: Failed to determine user credentials: No such process
Aug 17 12:34:05 gabriel systemd[12845]: tinc@tinc.service: Failed at step USER spawning /usr/bin/chown: No such process
Aug 17 12:34:05 gabriel systemd[1]: tinc@tinc.service: Control process exited, code=exited, status=217/USER
Aug 17 12:34:05 gabriel systemd[1]: tinc@tinc.service: Failed with result 'exit-code'.
Aug 17 12:34:05 gabriel systemd[1]: Failed to start Tinc net tinc.
```

**Comment by hg** [comment]
Try adding `ProtectSystem=no`.

**Comment by fangfufu** [comment]
With ``ProtectSystem=no``, I get: 
```
Aug 17 12:47:27 gabriel systemd[1]: Starting Tinc net tinc...
Aug 17 12:47:27 gabriel chown[61888]: /usr/bin/chown: invalid user: ‘tinc:tinc’
Aug 17 12:47:27 gabriel systemd[1]: tinc@tinc.service: Control process exited, code=exited, status=1/FAILURE
Aug 17 12:47:27 gabriel systemd[1]: tinc@tinc.service: Failed with result 'exit-code'.
Aug 17 12:47:27 gabriel systemd[1]: Failed to start Tinc net tinc.
```

**Comment by fangfufu** [comment]
I suppose I could fix it by adding the user and group ``tinc:tinc``, but then that requires changes to the installation script.

**Comment by hg** [comment]
Then it's running chown before creating the user, or maybe with a separate mount atop of `/etc/passwd`.

It's not worth it in any case, we'll have to adjust installation scripts first.

File: systemd/tinc@.service.in [hunk 1]
```diff
@@ -4,6 +4,9 @@ Documentation=info:tinc
 Documentation=man:tinc(8) man:tinc.conf(5)
 Documentation=http://tinc-vpn.org/docs/
 PartOf=tinc.service
+Before=systemd-user-sessions.service
+After=network-online.target
+Wants=network-online.target
```
Comment (Mic92) [inline systemd/tinc@.service.in:9]: Not in favor of this, see: https://github.com/gsliepen/tinc/pull/326#discussion_r711580446

File: systemd/tinc@.service.in [hunk 2]
```diff
@@ -4,6 +4,9 @@ Documentation=info:tinc
 Documentation=man:tinc(8) man:tinc.conf(5)
 Documentation=http://tinc-vpn.org/docs/
 PartOf=tinc.service
+Before=systemd-user-sessions.service
```
Comment (Mic92) [inline systemd/tinc@.service.in:7]: Not needed see: https://github.com/gsliepen/tinc/pull/326/files#r711580552

**Comment by fangfufu** [comment]
Closing as per https://github.com/gsliepen/tinc/pull/326#issuecomment-922352678

**Pull request closed without merging**